<a href="https://colab.research.google.com/github/carlosdev-ia/sns_curso2526/blob/main/tema2SNS.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## 🧮 Estadísticos descriptivos para una variable continua

Supongamos que los valores de una variable continua son  

$$
1, 2, 3, \ldots, 10.
$$

Queremos calcular:

- La **media aritmética (promedio)**  
- La **mediana**  
- La **desviación estándar muestral**  
- La **varianza muestral**  
- El **rango**

---

### 🔹 1. Media aritmética
$$
\bar{x} = \frac{\sum_{i=1}^{n} x_i}{n}
$$

---

### 🔹 2. Mediana

- Si \( n \) es **impar**:
  $$
  \text{Mediana} = x_{\frac{n+1}{2}}
  $$

- Si \( n \) es **par**:
$$
\text{Mediana} = \frac{x_{\left(\frac{n}{2}\right)} + x_{\left(\frac{n}{2} + 1\right)}}{2}
$$


---

### 🔹 3. Varianza muestral
$$
s^2 = \frac{\sum_{i=1}^{n} (x_i - \bar{x})^2}{n - 1}
$$

---

### 🔹 4. Desviación estándar muestral
$$
s = \sqrt{s^2} = \sqrt{\frac{\sum_{i=1}^{n} (x_i - \bar{x})^2}{n - 1}}
$$

---

### 🔹 5. Rango
$$
R = x_{\text{máx}} - x_{\text{mín}}
$$

---



## <br>Ejercicio 1
<br>

##Supongamos que los valores de una variable (continua) son 1, 2, 3,..., 10.   Calcula la media, la mediana, la desviacion estanadar muestral, la varianza muestral y el rango.

In [ ]:
import pandas as pd

# Importamos la librería numpy para poder generar datos aleatorios
# de una distribución normal de media y desviacióón estándar dada
# y para calcular estadísticos descriptivos básicos
  #import numpy as np

# Gráficos
  #import matplotlib.pyplot as plt

# Datos
n = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
df = pd.DataFrame(n, columns=["Valores"])

# Cálculos
media = df.mean()
mediana = df.median()
varianza = df.var()
desviacion_estandar = df.std()
rango = df.max() - df.min()

# Impresiones originales (con dtype)
print("\n📊 Impresiones originales")
print("Media:", media)
print("Mediana:", mediana)
print("Varianza:", varianza)
print("Desviación:", desviacion_estandar)
print("Rango:", rango)

# Impresiones limpias (sin dtype)
print("\n✅ Impresiones limpias sin dtype ni warnings")
print("Media:", media.iloc[0])
print("Mediana:", mediana.iloc[0])
print("Varianza:", varianza.iloc[0])
print("Desviación estándar:", desviacion_estandar.iloc[0])
print("Rango:", rango.iloc[0])



📊 Impresiones originales
Media: Valores    5.5
dtype: float64
Mediana: Valores    5.5
dtype: float64
Varianza: Valores    9.166667
dtype: float64
Desviación: Valores    3.02765
dtype: float64
Rango: Valores    9
dtype: int64

✅ Impresiones limpias sin dtype ni warnings
Media: 5.5
Mediana: 5.5
Varianza: 9.166666666666666
Desviación estándar: 3.0276503540974917
Rango: 9


## 🧮 Estadísticos descriptivos para una variable continua

Supongamos que los valores de la variable continua son  
\( 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 \)  
\((n = 10)\)

Queremos calcular:

- La **media aritmética**
- La **mediana**
- La **desviación estándar muestral**
- La **varianza muestral**
- El **rango**

---

### 🔹 1. Media aritmética
$$
\bar{x} = \frac{1 + 2 + 3 + 4 + 5 + 6 + 7 + 8 + 9 + 10}{10}
$$
<br>
$$
\bar{x} = \frac{55}{10} = 5.5
$$

---

### 🔹 2. Mediana
Como \( n = 10 \) (par), se toma el promedio de los valores en las posiciones  
\( \frac{n}{2} = 5 \) y \( \frac{n}{2} + 1 = 6 \):

$$
\text{Mediana} = \frac{x_{\left(\frac{n}{2}\right)} + x_{\left(\frac{n}{2} + 1\right)}}{2}
= \frac{x_{(5)} + x_{(6)}}{2}
= \frac{5 + 6}{2} = 5.5
$$

---

### 🔹 3. Varianza muestral
$$
s^2 = \frac{\sum_{i=1}^{10} (x_i - \bar{x})^2}{n - 1}
= \frac{(1-5.5)^2 + (2-5.5)^2 + \ldots + (10-5.5)^2}{9}
$$
<br>
$$
s^2 = \frac{82.5}{9} \approx 9.17
$$

---

### 🔹 4. Desviación estándar muestral
$$
s = \sqrt{s^2} = \sqrt{9.17} \approx 3.03
$$

---

### 🔹 5. Rango
$$
R = x_{\text{máx}} - x_{\text{mín}} = 10 - 1 = 9
$$

---

## <br>Ejercicio 2
<br>

##Supongamos que tenemos la muestra 1, 2, 3, . . . , 10. ¿Cómo se verían afectados la media, la mediana, la desviación estándar muestral, la varianza muestral y el rango si todos los datos de la muestra se multiplican por 2? ¿Y si en lugar de multiplicarlos por dos se le hubiera sumado 2 a cada uno de ellos?

In [17]:
import pandas as pd
import numpy as np

# Muestra original
x = np.arange(1, 11)  # [1, 2, ..., 10]

# Escenarios
x_multiplicado = x * 2
x_sumado = x + 2

# Función para calcular los estadísticos
def calculos_estadisticos(datos):
    return {
        "Media": np.mean(datos),
        "Mediana": np.median(datos),
        "Varianza": np.var(datos, ddof=1),
        "Desviación estándar": np.std(datos, ddof=1),
        "Rango": np.max(datos) - np.min(datos)
    }

# Cálculos
original = calculos_estadisticos(x)
multiplicado = calculos_estadisticos(x_multiplicado)
sumado = calculos_estadisticos(x_sumado)

# Crear DataFrame comparativo
tabla = pd.DataFrame([original, multiplicado, sumado],
                     index=["Original", "Multiplicado ×2", "Sumado +2"])

# Mostrar resultados con 3 decimales
print("📊 Comparativa de estadísticos\n")
print(tabla.round(3))


📊 Comparativa de estadísticos

                 Media  Mediana  Varianza  Desviación estándar  Rango
Original           5.5      5.5     9.167                3.028      9
Multiplicado ×2   11.0     11.0    36.667                6.055     18
Sumado +2          7.5      7.5     9.167                3.028      9


## 🧮 Efecto de las transformaciones lineales en los estadísticos

Supongamos que tenemos la muestra original:

$$
x = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
$$

De la muestra original ya sabemos que:

$$
\bar{x} = 5.5, \quad
\text{Mediana} = 5.5, \quad
s = 3.03, \quad
s^2 = 9.17, \quad
R = 9
$$

---

## 🔹 Escenario 1: Multiplicamos todos los valores por 2

Nueva muestra:
$$
x' = [2, 4, 6, 8, 10, 12, 14, 16, 18, 20]
$$

---

### 📘 Cálculos

**Media:**
$$
\bar{x} = \frac{2 + 4 + 6 + 8 + 10 + 12 + 14 + 16 + 18 + 20}{10}
= \frac{110}{10} = 11
$$

**Mediana:**
$$
\text{Mediana} = \frac{10 + 12}{2} = 11
$$

**Varianza muestral:**
$$
s^2 = \frac{\sum_{i=1}^{10} (x_i - 11)^2}{9}
= \frac{330}{9} = 36.67
$$

**Desviación estándar muestral:**
$$
s = \sqrt{36.67} = 6.06
$$

**Rango:**
$$
R = 20 - 2 = 18
$$

---

### 📊 Efecto en los estadísticos

- La **media**, la **mediana** y el **rango** se multiplican por 2.  
- La **varianza** se multiplica por \( 2^2 = 4 \).  
- La **desviación estándar** se multiplica por 2.

🟢 **Conclusión:**
$$
\begin{aligned}
\bar{x} &= 2 \times 5.5 = 11 \\
\text{Mediana} &= 11 \\
s^2 &= 4 \times 9.17 = 36.68 \\
s &= 2 \times 3.03 = 6.06 \\
R &= 2 \times 9 = 18
\end{aligned}
$$

---

## 🔹 Escenario 2: Sumamos 2 a todos los valores

Nueva muestra:
$$
x'' = [3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
$$

---

### 📘 Cálculos

**Media:**
$$
\bar{x} = \frac{3 + 4 + 5 + 6 + 7 + 8 + 9 + 10 + 11 + 12}{10}
= \frac{75}{10} = 7.5
$$

**Mediana:**
$$
\text{Mediana} = \frac{7 + 8}{2} = 7.5
$$

**Varianza muestral:**
$$
s^2 = \frac{\sum_{i=1}^{10} (x_i - 7.5)^2}{9}
= \frac{82.5}{9} = 9.17
$$

**Desviación estándar muestral:**
$$
s = \sqrt{9.17} = 3.03
$$

**Rango:**
$$
R = 12 - 3 = 9
$$

---

### 📊 Efecto en los estadísticos

- La **media** y la **mediana** aumentan en 2 unidades.  
- La **varianza**, **desviación estándar** y **rango** no cambian.
  Ya que todas las medidas de dispersión dependen de las diferencias entre los valores, no de su posición.

🟢 **Conclusión:**
$$
\begin{aligned}
\bar{x} &= 5.5 + 2 = 7.5 \\
\text{Mediana} &= 7.5 \\
s^2 &= 9.17 \\
s &= 3.03 \\
R &= 9
\end{aligned}
$$

---

### ✳️ Resumen comparativo

| Estadístico | Original | Multiplicado ×2 | Sumado +2 |
|--------------|-----------:|----------------:|-----------:|
| Media | 5.5 | 11.0 | 7.5 |
| Mediana | 5.5 | 11.0 | 7.5 |
| Varianza | 9.17 | 36.68 | 9.17 |
| Desviación estándar | 3.03 | 6.06 | 3.03 |
| Rango | 9 | 18 | 9 |


## 🧾 Conclusión general

El análisis muestra cómo las **transformaciones lineales** (multiplicar o sumar una constante) afectan a los principales **estadísticos descriptivos** de una muestra.

En la **muestra original** \( x = [1, 2, 3, \ldots, 10] \), los valores centrales y de dispersión fueron:

$$
\bar{x} = 5.5, \quad \text{Mediana} = 5.5, \quad s = 3.03, \quad s^2 = 9.17, \quad R = 9
$$
---

### 🔹 Escenario 1 – Multiplicar por 2
Cuando todos los valores se multiplican por 2, los estadísticos cambian de la siguiente forma:
- La **media**, la **mediana** y el **rango** se **duplican**.
- La **varianza** se **cuadruplica** (\( s^2 \times 4 \)).
- La **desviación estándar** también se **duplica**.

Esto ocurre porque al multiplicar cada dato por una constante \( k \), las distancias entre ellos también se multiplican por \( k \), afectando directamente a las medidas de dispersión.

$$
\bar{x}_{1} = 11, \quad s^2_{1} = 36.68, \quad s_{1} = 6.06, \quad R_{1} = 18
$$
---

### 🔹 Escenario 2 – Sumar 2
Cuando a cada valor se le suma una constante:
- La **media** y la **mediana** se **desplazan** en la misma cantidad (aumentan en 2 unidades).  
- Las **medidas de dispersión** (**varianza**, **desviación estándar** y **rango**) **no cambian**.

Esto se debe a que las distancias entre los datos permanecen iguales; solo se traslada la distribución en el eje de los valores.
$$
\bar{x}_{2} = 7.5, \quad s^2_{2} = 9.17, \quad s_{2} = 3.03, \quad R_{2} = 9
$$
---

### 📊 Conclusión global

- **Multiplicar** una muestra por una constante **afecta tanto la posición como la dispersión** de los datos.  
- **Sumar** una constante **solo desplaza** la distribución, **sin alterar su forma ni su dispersión**.

En síntesis:

> 🔸 Las **transformaciones multiplicativas** escalan la distribución.  
> 🔸 Las **transformaciones aditivas** la trasladan.  
> 🔸 Las **relaciones internas** entre los datos permanecen proporcionales en ambos casos.

Estas propiedades son fundamentales en estadística, ya que permiten **reajustar o estandarizar** conjuntos de datos sin modificar su estructura esencial.
